heartbeat/monitors/active/http/task.go

```diff
@@ -69,7 +61,17 @@ func newHTTPMonitorHostJob(
 	timeout := config.Timeout
 
 	return jobs.MakeSimpleJob(func(event *beat.Event) error {
-		_, _, err := execPing(event, client, request, body, timeout, validator, config.Response, &redirects)
+		var redirects []string
```

Comment:
I guess we use an array to ensure order. But maybe we should also check for redirect loops. Either by searching the redirects list or using a StringSet.

Reply:
We do require that `MaxRedirects` be set, so there's a limit inherent there. I think that's sufficient for loop detection IMHO, esp. since the default is zero. 

Experimenting with `cURL` it seems to just have a limit of 50 redirects, and not detect loops.